### DISCUSSION TITLE
Problems to understand add_callback.

### DISCUSSION THREAD

**Discussion by Wallat1** [bitcraze/discussions#687] [Q&A] [answered]
Hi.

I am having trouble understanding this part of the code `logconf.data_received_cb.add_callback(log_stab_callback)`,
 since I don't understand how using the above, it is not necessary to define the function arguments 
`def log_stab_callback(timestamp, data, logconf):`.
 Can you explain the above please? I leave the code below.

```
def log_stab_callback(timestamp, data, logconf):
    print('[%d][%s]: %s' % (timestamp, logconf.name, data))

def simple_log_async(scf, logconf):
    cf = scf.cf
    cf.log.add_config(logconf)
    logconf.data_received_cb.add_callback(log_stab_callback)
    logconf.start()
    time.sleep(5)
    logconf.stop()
```
Regards.

**Comment by ataffanel** [comment, marked as answer]
Hi!

The add_callback function takes a python function as an argument. In python, a function name can be used to get a function *object*, it can then be passed as argument or put in a variable, for  example:
``` python
>>> def hello(name):
...   print(f"hello {name}")
... 
>>> hello
<function hello at 0x7ff6fb502dd0>
>>> function = hello
>>> function("world")
hello world
>>> 
```

This is what is happening here, the function `log_stab_callback` is passed as an argument to `add_callback` so that it can be called each time a log block is received.

**Reply by Wallat1** [reply]
Hi.

Thank you so much for your help.

Regards.

**Comment by NageshEranki** [comment]


**Reply by knmcguire** [reply]
Could you start a new discussion? This one has already been answered and old so it's kind of 'closed'.
